Ginger/GingerCore/ALM/QtestCore.cs

```diff
@@ -0,0 +1,685 @@
+            catch (Exception ex)
+            {
+                result = "Unexpected error occurred- " + ex.Message;
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to export the " + GingerDicser.GetTermResValue(eTermResKey.BusinessFlow) + " to qTest", ex);
+                return false;
+            }
+        }
+
+        public QtestTest GetQtestTest(long? testID)
+        {
+            testcaseApi = new QTestApi.TestcaseApi(connObj.Configuration);
+            QTestApiModel.TestCaseWithCustomFieldResource testCase = testcaseApi.GetTestCase((long)Convert.ToInt32(ALMCore.DefaultAlmConfig.ALMProjectKey), testID);
+
+            QtestTest test = new QtestTest();
+            test.Description = testCase.Description;
+            test.TestName = testCase.Name;
+            test.TestID = testCase.Id.ToString();
+            foreach (QTestApiModel.TestStepResource testStep in testCase.TestSteps)
+            {
+                if (testStep.CalledTestCaseId != null)
+                {
+                    QTestApiModel.TestCaseWithCustomFieldResource calledTestCase = testcaseApi.GetTestCase((long)Convert.ToInt32(ALMCore.DefaultAlmConfig.ALMProjectKey), testStep.CalledTestCaseId);
+                    calledTestCase.TestSteps.ForEach(z => test.Steps.Add(new QtestTestStep(z.Id.ToString(), StripHTML(z.Description), StripHTML(z.Expected), testStep.CalledTestCaseId.ToString())));
+                }
+                else
+                {
+                    test.Steps.Add(new QtestTestStep(testStep.Id.ToString(), testStep.Description, testStep.Expected));
+                }
+            }
+              
+            return test;
+        }
+
+        public QtestTestSuite ImportTestSetData(QtestTestSuite TS)
+        {
+            ConnectALMServer();
+            testrunApi = new QTestApi.TestrunApi(connObj.Configuration);
+            testcaseApi = new QTestApi.TestcaseApi(connObj.Configuration);
+
+            List<QTestApiModel.TestRunWithCustomFieldResource> testRunList = testrunApi.GetOf((long)Convert.ToInt32(ALMCore.DefaultAlmConfig.ALMProjectKey), (long)Convert.ToInt32(TS.ID), "test-suite", "descendents");
+            foreach (QTestApiModel.TestRunWithCustomFieldResource testRun in testRunList)
+            {
+                QtestTest test = GetQtestTest(testRun.TestCase.Id);
+                test.Runs = new List<QtestTestRun>();
+                test.Runs.Add(new QtestTestRun(testRun.Id.ToString(), testRun.Name, testRun.Properties[0].ToString(), testRun.CreatorId.ToString()));
+                TS.Tests.Add(test);
+            }
+          
+            return TS;
+        }
+
+        public ObservableList<QTestApiModel.TestCycleResource> GetQTestCyclesByProject(string qTestServerUrl, string qTestUserName, string qTestPassword, string qTestProject)
+        {
+            ConnectALMServer();
+            ObservableList<QTestApiModel.TestCycleResource> cyclestList = new ObservableList<QTestApiModel.TestCycleResource>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'cyclestList'.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)